examples/colormap_normalization/colormap_normalizations_piecewisenorm.py

```diff
@@ -0,0 +1,103 @@
+"""
+============================================
+Examples of arbitrary colormap normalization
+============================================
+
+Here I plot an image array with data spanning for a large dynamic range,
+using different normalizations. Look at how each of them enhances
+different features.
+
+"""
+
+
+import matplotlib.cm as cm
+import matplotlib.colors as colors
+import matplotlib.pyplot as plt
+
+from mpl_toolkits.mplot3d import Axes3D
+
+import numpy as np
+
+from sampledata import PiecewiseNormData
+
+X, Y, data = PiecewiseNormData()
+cmap = cm.spectral
+
+# Creating functions for plotting
+
+
+def make_plot(norm, label=''):
```

Comment:
About the data, I do not think we can just have something easy and simple, because it needs to show the normalization in action, and we are dealing here with pretty sophisticated normalization.

Maybe I could split the examples in many files so each of them always compares the linear case to a different type of norm. But I also think the normalization line is very useful for the user to understand better what is happening. Maybe a (2,2), where the top axes correspond to the linear case (normalization line, and example image), and the bottom axes to the normalize case. In this case I could make fake data according to what I want to show on each of the examples, instead of using the same fake data for all. Nevertheless even in that case I will probably should make use of an external module to generate the fake data, like I am doing now. How would you feel about this?

> And I think the 3D should be its own example.

I am not sure what you mean exactly by that.